### PR TITLE
Enhancement (master) : Creating the Retrohunt interface.

### PR DESCRIPTION
**Pull request by cccs-nr**

### CODE REVIEW

File: src/components/routes/retrohunt/components/model.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  pending_indices: any;
+  pending_candidates: any;
+  errors: any;
+  hits: any;
+  finished: any;
+  truncated: any;
+  archive_only?: boolean;
+};
+
+export const DEFAULT_RETROHUNT: Retrohunt = {
```
Comment (cccs-sgaron) [inline src/components/routes/retrohunt/components/model.tsx:31]: There is a lot of default values that should not exist here

File: src/components/routes/retrohunt/components/json.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import Editor, { loader } from '@monaco-editor/react';
```
Comment (cccs-sgaron) [inline src/components/routes/retrohunt/components/json.tsx:1]: Why is this file called json.tsx when it's a monaco editor for yara rules?

File: src/components/routes/retrohunt/components/results.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+import { useTheme } from '@mui/material';
+import useAppTheme from 'commons/components/app/hooks/useAppTheme';
+import 'moment/locale/fr';
+import { useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import ReactResizeDetector from 'react-resize-detector';
+
+loader.config({ paths: { vs: '/cdn/monaco_0.35.0/vs' } });
+
+const yaraDef = {
```
Comment (cccs-sgaron) [inline src/components/routes/retrohunt/components/results.tsx:11]: You should use the yaraDef and yaraConfig from utils

**Review by cccs-sgaron — CHANGES_REQUESTED** [review]

File: src/components/routes/retrohunt/components/model.tsx (hunk 1 above)
Comment (cccs-nr) [reply src/components/routes/retrohunt/components/model.tsx]: I removed all the sub-files that I had before into one concise file that has all the duplicated values removed.

File: src/components/routes/retrohunt/components/json.tsx (hunk 2 above)
Comment (cccs-nr) [reply src/components/routes/retrohunt/components/json.tsx]: I added a single instance of monaco editor to be called and I removed the JSONEditor.

File: src/components/routes/search.tsx [hunk 4]
```diff
@@ -375,20 +391,23 @@ function Search({ index }: SearchProps) {
         </div>
       </PageHeader>
       <div style={{ paddingTop: theme.spacing(2), paddingLeft: theme.spacing(0.5), paddingRight: theme.spacing(0.5) }}>
-        {tab === 'submission' && query && query.get('query') && (
-          <SubmissionsTable submissionResults={submissionResults} allowSort={!!(index || id)} />
+        {tab === 'alert' && query && query.get('query') && (
```
[the remark this answers is not in the document]
Comment (cccs-nr) [reply src/components/routes/search.tsx:394]: I reverted that change to what it was before.

File: src/components/routes/retrohunt/retrohunt_detail.tsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router';
+import { useLocation, useParams } from 'react-router-dom';
+import { RetrohuntAdd, RetrohuntPageType, RetrohuntResults, RetrohuntView, RetrohuntYara } from './components';
+
+type ParamProps = {
+  code: string;
+};
+
+export type Retrohunt = {
```
[the remark this answers is not in the document]
Comment (cccs-nr) [reply src/components/routes/retrohunt/retrohunt_detail.tsx]: I put every type definition in the Retrohunt component page to remove the model definition.

**Comment by cccs-sgaron** [comment]
Dev branch was merged into this branch. Was replaced by: https://github.com/CybercentreCanada/assemblyline-ui-frontend/pull/740

**Pull request closed without merging**